services/server/src/lib/session.ts

```diff
@@ -64,46 +64,68 @@ export function canBook({
 }
 
 // todo: write tests
-export async function canJoinSession({
+/**
+ * this function has been written to clean up the wss session handler
+ * by seperating this verbose details from the main logic code.
+ */
+export async function canJoinSessionAck({
   userId,
   sessionId,
 }: {
   userId: number;
   sessionId: string;
-}) {
+}): Promise<Wss.AcknowledgePayload> {
   const now = dayjs.utc();
   const sessionType = first(sessionId.split(":"));
```

Comment:
It is better to move this function to `@litespace/sol`. 

```ts
function getSessionType(sessionId: ISession.Id): ISession.Type | null;
```
```suggestion
  const sessionType = getSessionType(sessionId)
```